dalston/block-extends/extend-cover-block.css

```diff
@@ -144,23 +144,27 @@
 	text-decoration: underline;
 }
 
-.wp-block-cover.is-style-hover-reveal .wp-block-cover__inner-container {
+.wp-block-cover.is-style-hover-reveal .wp-block-cover__inner-container,
+.wp-block-cover.is-style-hover-reveal .wp-block-cover-text {
 	opacity: 0;
 	-webkit-transition: opacity 0.2s ease-in-out;
 	-moz-transition: opacity 0.2s ease-in-out;
 	transition: opacity 0.2s ease-in-out;
 }
 
 .wp-block-cover.is-style-hover-reveal.dalston-in-viewport .wp-block-cover__inner-container,
-.wp-block-cover.is-style-hover-reveal:hover .wp-block-cover__inner-container {
+.wp-block-cover.is-style-hover-reveal.dalston-in-viewport .wp-block-cover-text,
+.wp-block-cover.is-style-hover-reveal:hover .wp-block-cover__inner-container,
+.wp-block-cover.is-style-hover-reveal:hover .wp-block-cover-text {
 	opacity: 1;
 }
 
 /* Dim */
 
 .wp-block-cover.is-style-hover-reveal.has-background-dim:before,
 .wp-block-cover.is-style-hover-reveal.has-background-dim:not(.has-background-gradient):before,
-.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-gradient .wp-block-cover__gradient-background {
+.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-gradient .wp-block-cover__gradient-background,
+.wp-block-cover.is-style-hover-reveal .wp-block-cover__background.has-background-dim {
```

Comment:
In the current Gutenberg version the `has-background-dim` class is applied to the element with `wp-block-cover__background` class and not to the `wp-block-cover`.